File: src/checkers-api/Models/GameModels/MoveResult.cs
namespace checkers_api.Models.GameModels;

public class MoveResult
{
    public string GameId { get; }
    public bool WasMoveSuccessful { get; }
    public bool IsGameOver { get; }
    public IEnumerable<Piece?> Board { get; }

    public MoveResult(string gameId, bool wasMoveSuccessful, bool isGameOver, IEnumerable<Piece?> board)
    {
        ArgumentNullException.ThrowIfNull(gameId);
        ArgumentNullException.ThrowIfNull(wasMoveSuccessful);
        ArgumentNullException.ThrowIfNull(isGameOver);
        ArgumentNullException.ThrowIfNull(board);

        GameId = gameId;
        WasMoveSuccessful = wasMoveSuccessful;
        Board = board;
        IsGameOver = isGameOver;
    }
}

File: src/checkers-client/src/game-models/moveResult.ts
import Square from "./square";

export default class MoveResult {
    public readonly gameid: string;
    public readonly wasmovesuccessful: boolean;
    public readonly isgameover: boolean;
    public readonly board: Square[];

    constructor(gameid: string, 
        wasmovesuccessful: boolean,
        isgameover: boolean,
        board: Square[]) {
        this.gameid = gameid;
        this.wasmovesuccessful = wasmovesuccessful;
        this.isgameover = isgameover;
        this.board = board;
    }
}
